@here/harp-map-controls/lib/MapControls.ts

```diff
@@ -1110,8 +1110,27 @@ export class MapControls extends THREE.EventDispatcher {
         const from = utils.calculateNormalizedDeviceCoordinates(fromX, fromY, width, height);
         const to = utils.calculateNormalizedDeviceCoordinates(toX, toY, width, height);
 
-        const toWorld = MapViewUtils.rayCastWorldCoordinates(this.mapView, to.x, to.y);
-        const fromWorld = MapViewUtils.rayCastWorldCoordinates(this.mapView, from.x, from.y);
+        let toWorld: THREE.Vector3 | null;
+        let fromWorld: THREE.Vector3 | null;
+        if (this.mapView.elevationProvider === undefined) {
+            fromWorld = MapViewUtils.rayCastWorldCoordinates(this.mapView, from.x, from.y);
+            toWorld = MapViewUtils.rayCastWorldCoordinates(this.mapView, to.x, to.y);
+        } else {
+            fromWorld = this.mapView.elevationProvider.rayCast(fromX, fromY);
+            if (fromWorld === null) {
+                return;
+            }
+            const geoCoordFromWorld = this.mapView.projection.unprojectPoint(fromWorld);
```

Comment:
You only need the altitude and that is a no-op for globe.
So maybe we should add a method to the projection that returns the altitude for a world coordinate

Reply:
You mean a no-op for planar projections?
Good idea to add a method to projection that returns the altitude. I will add it now and we can discuss the best name.
Renamed.